rust/impl/dt/type_constructor/derive_pair/src/lib.rs

```diff
@@ -0,0 +1,489 @@
+  {
+    let struct_name = &self.ident;
+    let generics = &self.generics;
+    let gtypes = &self.gtypes;
+
+    quote!
+    (
+      impl< #( #generics ),* > core::ops::DerefMut for #struct_name< #( #gtypes ),* >
+      {
+        #[ inline ]
+        fn deref_mut( &mut self ) -> &mut Self::Target
+        {
+          #[ cfg( debug_assertions ) ]
+          {
+            let layout1 = core::alloc::Layout::new::< Self >();
+            let layout2 = core::alloc::Layout::new::< Self::Target >();
+            debug_assert_eq!( layout1, layout2 );
+          }
+          /* Safety : in case of homopair it is safe to assume that layout is the same. Homopair does not have to have #[repr(C)]. */
+          unsafe
+          {
+            core::mem::transmute::< _, _ >( self )
+          }
+        }
+      }
+    )
+  }
+
+  fn impl_make0( &self ) -> proc_macro2::TokenStream
+  {
+    let struct_name = &self.ident;
+    let generics = &self.generics;
+    let gtypes = &self.gtypes;
+    let param1 = &self.fields.0.ty;
+    let param2 = &self.fields.1.ty;
+
+    quote!( _if_make!
+    {
+      impl< #( #generics ),* > Make0 for #struct_name< #( #gtypes ),* >
+      where
+        #param1 : Default,
+        #param2 : Default,
+      {
+        #[ inline ]
+        fn make_0() -> Self
+        {
+          Self( Default::default(), Default::default() )
+        }
+      }
+    })
+  }
+
+  fn impl_make1( &self ) -> proc_macro2::TokenStream
+  {
+    let struct_name = &self.ident;
+    let generics = &self.generics;
+    let gtypes = &self.gtypes;
+    let param = &self.fields.0.ty;
+
+    quote!( _if_make!
+    {
+      impl< #( #generics ),* > Make1< #param > for #struct_name< #( #gtypes ),* >
+      where
+        #param : Clone
+      {
+        #[ inline ]
+        fn make_1( val : #param ) -> Self
+        {
+          Self( val.clone(), val.clone() )
+        }
+      }
+    })
+  }
+
+  fn impl_make2( &self ) -> proc_macro2::TokenStream
+  {
+    let struct_name = &self.ident;
+    let generics = &self.generics;
+    let gtypes = &self.gtypes;
+    let param1 = &self.fields.0.ty;
+    let param2 = &self.fields.1.ty;
+
+    quote!( _if_make!
+    {
+      impl< #( #generics ),* > Make2< #param1, #param2 > for #struct_name< #( #gtypes ),* >
+      {
+        #[ inline ]
+        fn make_2( _1 : #param1, _2 : #param2 ) -> Self
+        {
+          Self( _1, _2 )
+        }
+      }
+    })
+  }
+}
+
+
+#[ proc_macro_derive( Pair ) ]
+pub fn derive_pair( input: proc_macro::TokenStream ) -> proc_macro::TokenStream
+{
+  let input = parse_macro_input!( input as syn::ItemStruct );
+  let dp = DerivePair::parse( input );
+
+  let mut impls : Vec< fn( dp : &DerivePair ) -> proc_macro2::TokenStream > = vec!
+  [
+    DerivePair::impl_to_tuple,
+    DerivePair::impl_clone_as_tuple,
+    DerivePair::impl_make0,
+    DerivePair::impl_make2,
+  ];
+  let impls_for_single_type : Vec< fn( dp : &DerivePair ) -> proc_macro2::TokenStream > = vec!
```

Comment:
Using function pointers instead of token trees is good optimization.

I think that items of this vector should be added to `impls` in the first branch of condition.